### DISCUSSION TITLE
Export as ES6 modules

### DISCUSSION THREAD

**Discussion by loxy** [codyhouse/codyhouse-framework#136] [Ideas]
Hi!

I really like your component lib! But it would be cool if there is an option to export the JavaScript as real ES6 modules. We are now in 2023 and IIFE feels not contemporary anymore. 

I have problems to import your stuff in an Astro page with this pattern. What I currently do is something like:

```js
// File#: _1_anim-menu-btn
// Usage: codyhouse.co/license
import { Util } from '../scripts';

export function AnimBtn() {
  var menuBtns = document.getElementsByClassName('js-anim-menu-btn');
  if (menuBtns.length > 0) {
    for (var i = 0; i < menuBtns.length; i++) {
      (function (i) {
        initMenuBtn(menuBtns[i]);
      })(i);
    }

    function initMenuBtn(btn) {
      btn.addEventListener('click', function (event) {
        console.log('anim button event');
        event.preventDefault();
        var status = !Util.hasClass(btn, 'anim-menu-btn--state-b');
        Util.toggleClass(btn, 'anim-menu-btn--state-b', status);
        // emit custom event
        var event = new CustomEvent('anim-menu-btn-clicked', { detail: status });
        btn.dispatchEvent(event);
      });
    }
  } else {
  }
}
```
And then call that in the component.

If I could select "as ES6" in export config would be very helpful. And I think this could be automated. I could even write a script for that.

Off topic:
TypeScript of course would be awesome, but too much work I guess. Or do you have any plans to re-work your JS foundation (e. g. transform `var` more to `const`)?

Best regards,
Kersten

**Comment by claudia-romano** [comment]
Hi Kersten,
thanks for the suggestion.
At the moment we do not have any plans for TypeScript. But we'll look into improving our Export options to include an ES6 config option as well. Cheers!

**Reply by asiejna-clarion** [reply]
This would be a great features!

**Reply by loxy** [reply]
Yeah, any news on that?

**Reply by claudia-romano** [reply]
Hi there, this is not a planned feature at the moment. But if you have a working script you want to share, please feel free to do so!  We can look into including it and see if it can be included in the export process! Thanks.
